### PR TITLE
Abstract and rename parser module related classes

### PR DESCRIPTION
**Pull request by Yikf**
### _Why are the changes needed?_

This pr aims to abstract and rename parser module related classes.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by Yikf** [comment]
cc @ulysses-you , Please take a look if you find a moment, i redefine the pacakge location:
image

File: kyuubi-server/src/main/scala/org/apache/kyuubi/sql/plan/trino/TrinoFeOperations.scala [hunk 1]
```diff
@@ -15,14 +15,14 @@
...
+package org.apache.kyuubi.sql.plan.trino
 
 import org.apache.kyuubi.sql.plan.KyuubiTreeNode
 
 /////////////////////////////////////////////////////////////////////////////////////////
 // This file contains all Trino JDBC operation nodes which are parsed from statement
 /////////////////////////////////////////////////////////////////////////////////////////
 
-case class TrinoGetSchemas(catalogName: String, schemaPattern: String) extends KyuubiTreeNode {
+case class GetSchemas(catalogName: String, schemaPattern: String) extends KyuubiTreeNode {
```
Comment (ulysses-you) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/sql/plan/trino/TrinoFeOperations.scala:26]: does this change necessary ? Seems Trinoxxx is more cleaer to me

File: kyuubi-server/src/main/scala/org/apache/kyuubi/sql/parser/KyuubiParserBase.scala [hunk 2]
```diff
@@ -20,49 +20,34 @@ package org.apache.kyuubi.sql.parser
...
-        parser.reset()
-
-        // Try Again.
-        parser.getInterpreter.setPredictionMode(PredictionMode.LL)
-        toResult(parser)
+  private def parseTree(parser: P): ParseTree = {
+    parser match {
+      case _: KyuubiSqlBaseParser => parser.asInstanceOf[KyuubiSqlBaseParser].singleStatement()
+      case _: KyuubiTrinoFeBaseParser =>
+        parser.asInstanceOf[KyuubiTrinoFeBaseParser].singleStatement()
```
Comment (ulysses-you) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/sql/parser/KyuubiParserBase.scala:46]: we can define a parse method then we do not need to match implementation in abstract class

File: kyuubi-server/src/main/scala/org/apache/kyuubi/sql/plan/trino/TrinoFeOperations.scala (hunk 1 above)
Comment (Yikf) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/sql/plan/trino/TrinoFeOperations.scala:26]: Kyuubi supports trino **engine** and trino **fe**, Trinoxxx seems to be ambiguous, or rename to TrinoFexxx?

**Comment by ulysses-you** [comment]
thanks, merging to master

**Pull request closed without merging**